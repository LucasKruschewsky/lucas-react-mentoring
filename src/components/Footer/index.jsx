import React from 'react';
import AppLogo from 'Components/AppLogo';
import FooterContainer from './styles';

export default function Footer() {
  return (
    <FooterContainer>
      <AppLogo />
    </FooterContainer>
  );
}
