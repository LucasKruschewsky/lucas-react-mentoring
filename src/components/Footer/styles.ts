import styled from 'styled-components';

const FooterContainer = styled.div`
  display: flex;
  justify-items: center;
  align-items: center;
  text-align: center;

  margin-top: 5rem;
  padding: 1rem;

  background-color: rgb(var(--secondary-light));

  & > * {
    width: 100%;
  }
`;

export { FooterContainer };
