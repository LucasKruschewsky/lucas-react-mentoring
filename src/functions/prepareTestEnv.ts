import { unmountComponentAtNode } from 'react-dom';

const prepareTestEnv = (): void => {
  let container: HTMLDivElement = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
};

export default prepareTestEnv;
