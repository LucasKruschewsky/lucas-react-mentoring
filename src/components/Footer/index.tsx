import * as React from 'react';
import AppLogo from 'Components/AppLogo';
import { FooterContainer } from './styles';

export default function Footer(): React.ReactElement {
  return (
    <FooterContainer>
      <AppLogo />
    </FooterContainer>
  );
}
