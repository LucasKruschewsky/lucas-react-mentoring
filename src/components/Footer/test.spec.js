import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import prepareTestEnv from 'Root/functions/prepareTestEnv';
import Footer from '.';

describe('Footer', () => {
  prepareTestEnv();

  it('Snapshot test', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
