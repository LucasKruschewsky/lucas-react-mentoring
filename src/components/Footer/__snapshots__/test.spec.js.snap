// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot test 1`] = `[Function]`;
